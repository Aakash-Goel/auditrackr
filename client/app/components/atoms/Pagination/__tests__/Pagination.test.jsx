import React from 'react';
import TestRenderer from 'react-test-renderer';

import Pagination from '../Pagination';

describe('<Pagination/>', () => {
  it('renders correctly', () => {
    const paginationWrapper = TestRenderer.create(
      <Pagination
        pages={[
          { text: 1 },
          { text: '...' },
          { active: true, text: 5 },
          { text: 6 },
          { text: 7 },
          { text: '...' },
          { text: 11 },
        ]}
      />
    ).toJSON();
    expect(paginationWrapper).toMatchSnapshot();
  });

  it('renders correctly when onclick props is passed', () => {
    const onClick = jest.fn();
    const paginationWrapper = TestRenderer.create(
      <Pagination
        pages={[
          { text: 1 },
          { text: '...' },
          { active: true, text: 5 },
          { onClick, text: 6 },
          { text: 7 },
          { text: '...' },
          { text: 11 },
        ]}
      />
    ).toJSON();
    expect(paginationWrapper).toMatchSnapshot();
  });
});
