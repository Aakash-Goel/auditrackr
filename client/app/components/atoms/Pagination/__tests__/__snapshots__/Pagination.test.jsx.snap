// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> renders correctly 1`] = `
<ul
  className="Pagination-pagination-1"
>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        1
      </span>
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        ...
      </span>
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 Pagination-primary-4 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        5
      </span>
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        6
      </span>
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        7
      </span>
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        ...
      </span>
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        11
      </span>
    </button>
  </li>
</ul>
`;

exports[`<Pagination/> renders correctly when onclick props is passed 1`] = `
<ul
  className="Pagination-pagination-1"
>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        1
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        ...
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 Pagination-primary-4 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        5
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        6
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        7
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        ...
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </li>
  <li
    className="Pagination-paginationItem-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root Pagination-paginationLink-3 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        11
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </li>
</ul>
`;
